src/coreclr/jit/gentree.cpp

```diff
@@ -19388,6 +19388,13 @@ bool GenTree::isRMWHWIntrinsic(Compiler* comp)
             // in the case `op3` is a constant and none of the nibbles are
             // `0`, then we don't have to be RMW and can actually "drop" `op1`
 
+            GenTree* op4 = hwintrinsic->Op(4);
+
+            if (!op4->IsIntegralConst())
+            {
+                return true;
+            }
+
```

Comment:
It was "missing". It's not actually hittable since we should only get here for the `mustExpand` path, in which case `op3` can't be a constant (non-inlined parameter). But, it's good to have the correct handling so bugs don't sneak in when we eventually are able to move the decision to `keep as intrinsic` or `keep as a call so a jump table can be generated` out of importation and into rationalization or lowering.